apps/docs-snippets/src/guide/encoding/encode-and-decode.test.ts

```diff
@@ -0,0 +1,123 @@
+import { generateTestWallet } from '@fuel-ts/account/test-utils';
+import {
+  FUEL_NETWORK_URL,
+  Provider,
+  AbiCoder,
+  FunctionInvocationResult,
+  Script,
+  ReceiptType,
+  arrayify,
+} from 'fuels';
+import type { Account, JsonAbi, JsonAbiArgument, TransactionResultReturnDataReceipt } from 'fuels';
+
+import abiSnippet from '../../../test/fixtures/abi/encode-and-decode.jsonc';
+import { SumScriptAbi__factory as factory } from '../../../test/typegen/scripts/factories/SumScriptAbi__factory';
+
+/**
+ * @group node
+ */
+describe('encode and decode', () => {
+  let wallet: Account;
+
+  beforeAll(async () => {
+    const provider = await Provider.create(FUEL_NETWORK_URL);
+    const assetId = provider.getBaseAssetId();
+    wallet = await generateTestWallet(provider, [{ assetId, amount: 500_000 }]);
+  });
+
+  it('generates valid ABI', () => {
+    expect(abiSnippet).toEqual(factory.abi);
+  });
+
+  it('encodes and decodes', async () => {
+    // #region encode-and-decode-3
+    // #import { JsonAbi, Script };
+    // #context import { factory } from './sway-programs-api';
```

Comment:
Is this line valid? 🤔 

I don't remember us having a `factory` property there.

Reply:
Should probably be something like [this](https://github.com/FuelLabs/fuels-ts/pull/2405/files#diff-e3d0c6c1264fe3529431371fe4d50cd6ef11a4b1baabbc774600053db5c764ccR14):

```ts
import { SumScriptAbi__factory as factory } from '../../../test/typegen/scripts/factories/SumScriptAbi__factory';
```

Reply:
The `factory` property has been there for some time but the import was missing, which I added as a previous review highlighted the missing import.

I opted for the `import { factory } from './sway-programs-api';` as it seems more intuitive given the typegen examples, and our script name and factory file path don't have relevance to the user.